/*
Copyright (C) 2021 Srivats P.

This file is part of "Ostinato"

This is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>
*/

import "protocol.proto";

package OstProto;

// GRE Protocol
message Gre {
    optional uint32    flags = 1 [default = 0xa];
    optional uint32    rsvd0 = 2;
    optional uint32    version = 3;
    optional uint32    protocol_type = 4;
    optional uint32    checksum = 5;
    optional uint32    rsvd1 = 6;
    optional uint32    key = 7 [default = 0x2020bad7];
    optional uint32    sequence_num = 8;
}

extend Protocol {
    optional Gre gre = 405;
}
